test/test_in_toto_run.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+"""
+<Program Name>
+  test_in_toto_run.py
+
+<Author>
+  Lukas Puehringer <lukas.puehringer@nyu.edu>
+
+<Started>
+  Nov 28, 2016
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Test in_toto_run command line tool.
+
+"""
+
+import os
+import sys
+import unittest
+import logging
+import argparse
+import shutil
+import tempfile
+from mock import patch
+
+from in_toto.util import (generate_and_write_rsa_keypair,
+    prompt_import_rsa_key_from_file)
+
+from in_toto.models.link import Link
+from in_toto.in_toto_run import main as in_toto_run_main
+from in_toto.in_toto_run import in_toto_run
+from in_toto.runlib import FILENAME_FORMAT
+
+
+
+WORKING_DIR = os.getcwd()
+
+# Suppress all the user feedback that we print using a base logger
+logging.getLogger().setLevel(logging.CRITICAL)
+
+class TestInTotoRunTool(unittest.TestCase):
+  """Test in_toto_run's main() - requires sys.argv patching; and
+  in_toto_run- calls runlib and error logs/exits on Exception. """
+
+  @classmethod
+  def setUpClass(self):
+    """Create and change into temporary directory,
+    generate key pair, dummy artifact and base arguments. """
+    self.test_dir = tempfile.mkdtemp()
+    os.chdir(self.test_dir)
+
+    self.key_path = "test_key"
+    generate_and_write_rsa_keypair(self.key_path)
+    self.key = prompt_import_rsa_key_from_file(self.key_path)
+
+    self.test_step = "test_step"
+    self.test_link = FILENAME_FORMAT.format(step_name=self.test_step)
+    self.test_artifact = "test_artifact"
+    open(self.test_artifact, "w").close()
+
+  @classmethod
+  def tearDownClass(self):
+    """Change back to initial working dir and remove temp test directory. """
+    os.chdir(WORKING_DIR)
+    shutil.rmtree(self.test_dir)
+
+  def tearDown(self):
+    try:
+      os.remove(self.test_link)
+    except OSError:
+      pass
+
+
+  def test_main_required_args(self):
+    """Test CLI command with required arguments. """
+
+    args = [ "in_toto_run.py", "--step-name", self.test_step, "--key",
+        self.key_path, "--", "echo", "test"]
+    with patch.object(sys, 'argv', args):
+      in_toto_run_main()
+
+    self.assertTrue(os.path.exists(self.test_link))
+
+
+  def test_main_optional_args(self):
+    """Test CLI command with optional arguments. """
+
+    args = [ "in_toto_run.py", "--step-name", self.test_step, "--key",
+        self.key_path, "--materials", self.test_artifact, "--products",
+        self.test_artifact, "--record-byproducts", "--", "echo", "test"]
+
+    with patch.object(sys, 'argv', args):
+      in_toto_run_main()
```

Comment:
I know that checking that there was a file written or not is better than checking that a link metadata object is not returned, but maybe we could do both?